modules/patterns/pages/managing-pipeline-changes.adoc

```diff
@@ -0,0 +1,125 @@
+
+Understanding this evolution helps teams choose the right approach for their needs at different stages, balancing simplicity, control, and reusability.
+
+.Use Cases
+* Initial onboarding to {ProductName} and understanding the default pipeline setup.
+* Gaining control and customizing build logic for components within a single repository.
+* Reducing duplication when multiple components in the same repository can share a common build pipeline.
+* Standardizing a customized build pipeline across multiple repositories by referencing a common Git source.
+* Gradually adopting more complex CI/CD practices by starting with a simple setup and evolving as needed.
+
+.Phases of Pipeline Management Evolution
+
+[[phase-1-initial-onboarding-and-default-setup]]
+== Phase 1: Initial Onboarding and Default Setup
+
+When you initially onboard a component to {ProductName}, the system typically configures `PipelineRun` files (in the `.tekton/` directory of your repository). These `PipelineRun` files define how builds are triggered and executed. Initially, they will contain an **embedded `pipelineSpec`**. The entire pipeline logic (parameters, tasks, etc.) is defined directly within each `PipelineRun` file. {ProductName} generates separate `PipelineRun` definitions for different events (e.g., one for push events to the target branch, another for pull request events). This means the same pipeline logic will be duplicated in both the push and pull request `PipelineRun` files for each component.
+
+.Characteristics:
+* **Setup**: Largely automated by {ProductName}.
+* **Maintenance**:
+    * The `pipelineSpec` is embedded in the `PipelineRun`, you directly see the logic, but it's duplicated across files.
+    * Mintmaker will propose duplicate updates to both `PipelineRun` files for every Component.
+* **Simplicity**: Quickest way to get a build running with {ProductName}'s default behavior.
+
+.Benefits:
+* Rapid component onboarding.
+
+.Considerations:
+* **Embedded Specs & Duplication**: If `pipelineSpec` is embedded, the same pipeline logic is duplicated across multiple `PipelineRun` files (e.g., for push and pull request events for each component). This makes maintenance difficult, as any change to the pipeline logic needs to be replicated in all relevant files. This duplication is a strong motivator for moving to Phase 2.
+
+[[phase-2-centralizing-pipeline-definition-locally-within-a-repository]]
+== Phase 2: Centralizing the Pipeline Definition Locally (Within a Repository)
+
+As build requirements become more specific, or to manage common pipeline logic for multiple components within the *same repository* and eliminate duplication, the next step is often to centralize the pipeline definition into a local file. This aligns with the pattern xref:centralize-pipeline-definitions.adoc[Centralizing Tekton Pipeline Definitions in Your Repository].
+
+.Procedure:
+. **Identify Common/Base Pipeline Logic**:
```

Comment:
Should this just xref link to the other file on centralizing pipeline definitions?

Reply:
I wanted to explain the overall approach in a single file first and then go in more details later. Maybe I am going into too much detail here?